drivers/char/rust_example.rs

```diff
@@ -45,13 +55,38 @@ struct RustExample {
     _dev: Pin<Box<miscdev::Registration>>,
 }
 
+#[cfg(CONFIG_SYSFS)]
+fn print_params() -> KernelResult<()> {
+    let lock = THIS_MODULE.kernel_param_lock();
+    println!("Parameters:");
+    println!("  my_bool:    {}", my_bool.nonlocking_read());
+    println!("  my_i32:     {}", my_i32.locking_read(&lock));
```

Comment:
This shouldn't be on the user side; otherwise all users will need to use conditional compilation on `CONFIG_SYSFS`. It is better if we do it once and keep the same API, like the C side does, i.e. we define the locking function to be a noop, `KParamGuard` a dummy unit type, etc.

I'd also use the `read()` name for all cases (i.e. with and without lock), since the functions will always be different anyway (so it won't compile if the user makes a mistake) and since one cannot know if there will be locking or not:

```rust
my_readonly_param.read();
my_writable_param.read(&lock); // may lock or not depending on `CONFIG_SYSFS`
```

Reply:
Made locking/unlocking a noop, but kept `KParamGuard` the same.